bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -258,6 +315,31 @@ private VolumeInfo getVolume(int stream) throws AndroidDebugBridgeDeviceExceptio
         return volumeInfo;
     }
 
+    public String recordInputEvents()
+            throws AndroidDebugBridgeDeviceException, InterruptedException, TimeoutException, ExecutionException {
+        String out = runAdbShell(recordDuration * 2, "getevent", "&", "sleep", Integer.toString(recordDuration), "&&",
+                "exit");
+        var matcher = INPUT_EVENT_PATTERN.matcher(out);
+        var commandList = new ArrayList<String>();
+        while (matcher.find()) {
+            String inputPath = matcher.group("input");
+            int n1 = Integer.parseInt(matcher.group("n1"), 16);
+            int n2 = Integer.parseInt(matcher.group("n2"), 16);
+            int n3 = Integer.parseInt(matcher.group("n3"), 16);
```

Comment:
Maybe you should catch parse exception

Reply:
I don't think is needed because the regex is quite explicit and the service will not return other values in that place other than alphanumerical numbers. 
Either way I can add it if you want but wrapping the complete while because the command recorded have no sense if we miss a part. 
Let me know what I should do.

Reply:
Your pattern matching does not guarantee that your string will match a hexa number. This can be any string without white spaces.
So either your update your pattern to be sure to match a hexa number and you do not need catching exception, or your keep your pattern unchallenged and it will be safer to catch exception.

Reply:
I have added a NumberFormatException catch around the while loop.